pkg/skaffold/build/options.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+)
+
+// Configuration denotes build configuration for the artifact builder as either Release or Debug
+type Configuration int
+
+const (
+	Release = iota
+	Debug
+)
+
+// ImageOptions provides options for the artifact builder
+type ImageOptions struct {
+	// FIXME: Tag should be []string but we don't support multiple tags yet
+	Tag           string             // image tag
+	Configuration Configuration      // build image for release or debug
+	Args          map[string]*string // additional builder specific args
+}
+
+var CurrentConfiguration Configuration
+
+func CreateBuilderOptions(tag string) *ImageOptions {
+	return &ImageOptions{
+		Tag:           tag,
+		Configuration: CurrentConfiguration,
+	}
+}
```

Comment:
We really should avoid global state.  [`SkaffoldOptions.Command`](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/config/options.go#L76) records the command used on the command-line, and offers [`IsDevMode()` and `IsDebugMode()`](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/config/options.go#L92-L98) to test.

I wonder if it would be worth broadening the scope of this structure to capture the build options, including the artifact, and have it be passed in and around.

Reply:
Done